File: ClientApp/src/types/PlayerSessionDTO.ts
export interface PlayerSessionDTO {
  id: string;
  playerId: string;
  serverId: string;
  serverName: string;
  joinDate: string; // ISO string
  leaveDate?: string; // ISO string or undefined if active
  ipAddress: string;
  playerName: string;
  durationMinutes?: number;
  isActive: boolean;
}

File: GraniteServerMod/Api/Models/PlayerSessionDTO.cs
using System;

namespace GraniteServer.Api.Models;

public class PlayerSessionDTO
{
    public Guid Id { get; set; }
    public string PlayerId { get; set; } = string.Empty;
    public Guid ServerId { get; set; }
    public string ServerName { get; set; } = string.Empty;
    public DateTime JoinDate { get; set; }
    public DateTime? LeaveDate { get; set; }
    public string IpAddress { get; set; } = string.Empty;
    public string PlayerName { get; set; } = string.Empty;

    public double? Duration { get; set; }

    public bool IsActive { get; set; }
}
